drivers/storage/vbox/vbox_driver.go

```diff
@@ -0,0 +1,658 @@
+		if err != nil {
+			return nil, err
+		}
+
+		var bdn string
+		var ok bool
+		if bdn, ok = mapDiskByID[smid[0]]; !ok {
+			continue
+		}
+		sdBlockDevice := &types.Volume{
+			Name:   bdn,
+			ID:     mid,
+			Status: location,
+		}
+		blockDevices = append(blockDevices, sdBlockDevice)
+
+	}
+	return blockDevices, nil
+}
+
+func (d *Driver) VolumeInspect(
+	ctx context.Context,
+	volumeID string,
+	opts *drivers.VolumeInspectOpts) (*types.Volume, error) {
+	return nil, nil
+}
+
+func (d *Driver) VolumeCreate(
+	ctx context.Context,
+	name string,
+	opts *drivers.VolumeCreateOpts) (*types.Volume, error) {
+
+	if opts.Size == nil {
+		return nil, goof.New("missing volume size")
+	}
+
+	fields := map[string]interface{}{
+		"provider":   Name,
+		"volumeName": name,
+		"size":       *opts.Size,
+	}
+
+	size := *opts.Size * 1024 * 1024 * 1024
+
+	d.refreshSession()
+	volumes, err := d.GetVolume(ctx, "", name)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(volumes) > 0 {
+		return nil, goof.WithFields(fields, "volume exists already")
+	}
+
+	volume, err := d.createVolume(name, size)
+	if err != nil {
+		return nil, goof.WithFieldsE(fields, "error creating new volume", err)
+	}
+
+	// double check
+	volumes, err = d.GetVolume(ctx, volume.ID, "")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(volumes) == 0 {
+		return nil, goof.New("failed to get new volume")
+	}
+
+	newVol := &types.Volume{
+		ID:   volume.ID,
+		Name: volume.Name,
+		Size: volume.Size,
+		IOPS: *opts.IOPS,
+		Type: string(volume.DeviceType),
+	}
+
+	return newVol, nil
+}
+
+func (d *Driver) VolumeCreateFromSnapshot(
+	ctx context.Context,
+	snapshotID, volumeName string,
+	opts types.Store) (*types.Volume, error) {
+	return nil, types.ErrNotImplemented
+}
+
+func (d *Driver) VolumeCopy(
+	ctx context.Context,
+	volumeID, volumeName string,
+	opts types.Store) (*types.Volume, error) {
+	return nil, nil
+}
+
+func (d *Driver) VolumeSnapshot(
+	ctx context.Context,
+	volumeID, snapshotName string,
+	opts types.Store) (*types.Snapshot, error) {
+	return nil, types.ErrNotImplemented
+}
+
+func (d *Driver) VolumeRemove(
+	ctx context.Context,
+	volumeID string,
+	opts types.Store) error {
+
+	d.Lock()
+	defer d.Unlock()
+	d.refreshSession()
+
+	fields := map[string]interface{}{
+		"provider": Name,
+		"volumeID": volumeID,
+	}
+
+	err := d.vbox.RemoveMedium(volumeID)
+	if err != nil {
+		return goof.WithFieldsE(fields, "error deleting volume", err)
+	}
+
+	return nil
+}
+
+// TODO  - should VolumeAttach return an attachment ?
+func (d *Driver) VolumeAttach(
+	ctx context.Context,
+	volumeID string,
+	opts *drivers.VolumeAttachByIDOpts) (*types.Volume, error) {
+
+	if volumeID == "" {
+		return nil, errors.ErrMissingVolumeID
+	}
+
+	volumes, err := d.GetVolume(ctx, volumeID, "")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(volumes) == 0 {
+		return nil, errors.ErrNoVolumesReturned
+	}
+
+	// TODO - force flag from RR not ported here, see RR/drivers/storage/virtualbox.go
```

Comment:
Hi @clintonskitson,

If it's not ID/name related it should be in opts. I imagine I started this work before you added the force bit, and I had not backported it from your changes yet.